### ISSUE TITLE
CIF Input gathering for Amendments/Revisions workflows

### ISSUE THREAD

**Issue by nanyangpro** [bcgov/cas-cif#1422]
#### Describe the task
Gaps between the front-end business logic and the back-end technical logic have been identified and discussed with the developers as outcome of #1365. However, the workflow design have diverged into 2 main different directions:

1. When viewing/editing a revision record, other forms from the tasklist always show the current status of the project, i.e. always show the latest committed information
- [View Revision Record (Updated)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=3068%3A2675&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=3068%3A2675&show-proto-sidebar=1)
- [New Revision Record (Updated)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=3169%3A2715&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=3169%3A2715&show-proto-sidebar=1)
- [Edit Revision Record (Updated)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=7283%3A870&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=7283%3A870&show-proto-sidebar=1)

2. When viewing/editing a revision record, other forms from the tasklist are accessible and show a past status of the project, i.e. show information at the time of the revision record
- [Alternative Workflow](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=7314%3A3733&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=7314%3A3733&show-proto-sidebar=1)

**We need to choose one of the direction to move forward.** To make informed decision, CIF's input is required as suggested by the developers. This ticket is to discover and discuss confusions the CIF team may have with the updated interactive prototypes, and make design decisions based on them.

#### Acceptance Criteria
- [x] Write agenda and meeting invite for the input gathering session with CIF
- [x] Schedule the input gathering session with CIF
- [x] Draft test scripts and draft 
- [x] Conduct the usability test / input gathering session with CIF
- [x] Compile and document design decisions based on the input with the team
- [ ] Update the prototypes to reflect the design decisions we agree upon and remove older versions to avoid confusions

**Comment by nanyangpro** [comment]
Helpful comments in #1275.

**Comment by mikevespi** [comment]
Questions copied over from Teams:

If the decision is to go with the new flow, then answers to the following should help unblock some of the work. Numbers 1 and 2 were just points that were causing confusion and final decisions weren't made at the dev/design sync. Number three is a bigger issue that in my opinion needs feedback from the users.

1. Are we keeping the task list on the page with the holistic change log? There was confusion around this at the last dev/design sync, and no official decision was made.

2. Are we keeping the link to the change log in the task list, removing it, or separating it from the other items with a gap to indicate that it isn’t tied to a specific revision?

3. With General revisions and Minor revisions (a.k.a. routinely updated fields) being able to update the same set of fields, we’ve introduced the possibility of a non-linear history, which to date we’ve placed a lot of emphasis on maintaining. This is best understood with an example:

I start a General revision on a project, and one of the changes I make is setting the project status to “Under Technical Review”. I’m waiting to hear back from the proponent about something so I leave the revision `pending`.

Now you go in and change change the status to “Technical Review Complete” via a routinely updated field. The next day, I hear back from the client and finally submit my General revision. Your change is overwritten and the project status is now “Under Technical Review”.  Alternatively, when you hit submit on your routinely updated field, the status in my general revision is changed to “Technical Review Complete”, and now something has changed in my revision that wasn’t my doing. In this specific scenario, the latter is fine. There are others where the opposite is true. It’s a case by case basis which would be correct.

We all agree the likelihood of this happening is quite low, but not impossible, and we should avoid introducing the potential for it.


 

Do we need to give them two different ways to update the same set of fields? If so, I think a meeting with the ops team will be necessary because while there are ways to handle this, they are a little bit unintuitive and we need to make sure that any conflicts are handled the way the team expect them to be.

If however the CIF team feel that just one way to update that set of fields is no loss to them, then eliminating either General or Minor revisions, then we bypass the problem entirely. I lean towards General getting the axe if this is the route we take, as without them, our existing flow would need very little tweaking to accommodate for having Minor and amendments as the two types of revision.

**Comment by nanyangpro** [comment]
@LindsayMacfarlane's feedback:
**1. Are we keeping the task list on the page with the holistic change log? There was confusion around this at the last dev/design sync, and no official decision was made.**
For clarification, this is reference to when a user is viewing a revision? If so, then I would recommend removing the task list when a user is just looking at a previous revision. Allowing them to jump around is confusing and adds unnecessary complexity. 

**2. Are we keeping the link to the change log in the task list, removing it, or separating it from the other items with a gap to indicate that it isn’t tied to a specific revision?**
I feel I need to see an image. I would say keep the link in the task list. To me it already seems intuitive that you would select the amendments change log from the task list to see all amendments/revisions related to a particular project.

**3. With General revisions and Minor revisions (a.k.a. routinely updated fields) being able to update the same set of fields, we’ve introduced the possibility of a non-linear history, which to date we’ve placed a lot of emphasis on maintaining.**
Is part of the confusion here that we have general and minor revisions? Is this  adding in some of the complexity? Can we nudge the team into just two categories - Amendments and Revisions? This probably doesn't answer your specific question. Wouldn't the change that was 'committed' first be the first in this linear scenario? I always appreciate the team looking at all the what if scenarios, but in this context can we address at a later date if there is an issue or the scenario arises?

**Comment by nanyangpro** [comment]
Updates for @LindsayMacfarlane: 

- Usability testing and input gathering session was held with Amanda, Lindsay (CIF), @mikevespi, @BCerki, @suhafa, and @Dianadec on Thu, Feb 16.
- Prototypes of both the refined workflow and the alternative workflow was walked through by Amanda completing key user tasks: view/create/update Amendments/Revisions.
- Very helpful input was gathered. Questions from the developers were answered. We are in a good place to agree on design decisions to unblock associated development tickets. **Key take-aways below:**
    - After clicking on a form from the task list, user always expects to see the current status (latest committed information) of a project. _(With only one exception, see the next point below.)_
    - When creating a new Amendment, user expects to use the task list to navigate among forms to make edits.
    - The users see little value of keeping General Revision, given the technical complexity of avoiding potential duplications/conflicts.
    - As suggested, we'll only keep **Amendment** (created from the holistic change log table page) and **Minor Revision** (routinely updated fields).
    - As suggested, the "New Revision" button on the holistic change log table page should be relabeled as "New Amendment".
- A 45-minute briefing session is scheduled on Fri, Feb 17 afternoon for the attendees to sync-up on notes, confirm and document the agreed solutions to make sure the development tickets are all unblocked.

**Comment by nanyangpro** [comment]
Hi @LindsayMacfarlane - FYI, key design decisions compiled and agreed upon with @mikevespi, @BCerki, @suhafa, and @Dianadec are listed below:

- After clicking on a form from the task list, the user always expects to see the CURRENT status (latest committed information) of a project. (With only one exception, see the next point below.) 
- When creating a new Amendment or editing an open Amendment, the user expects to use the task list to navigate among forms to make edits. 
- The users see little value in the current workflow of creating a General/Minor Revision, given the technical complexity of avoiding potential duplications/conflicts. 
- As suggested, we will only keep the workflow of creating an Amendment from the holistic change log table page, and use the current workflow of updating routinely updated fields to create General Revisions. 
- There will be no more Minor Revisions. 
- Rename the "New Revision" button on the holistic change log table page to "New Amendment". 
- For an open Amendment, we will add more clarity to the “Forms updated” section so that the user is aware that by clicking the form title, they can navigate to it to make further edits. 
- Nice to have (low priority) - A “View History” button from the “Projects” page that brings the user directly to the holistic change log table page of a project. 
- With the page title, e.g. “Amendment 4”, “General Revision 2”, there will be no more “Revision type” section. 

**Next steps:** 

- Update the spreadsheet to reflect the new grouping of fields as discussed. 
- Refine the tickets for Amendments/Revisions accordingly to unblock implementation 
- Update the prototypes to reflect the design decisions. (Not a blocker of the implementation but can provide additional guidance according to Brianna.) 
- Update the business workflow charts to reflect the design decisions. (May or may not be needed, based on the upcoming updates from Mike)

**Comment by nanyangpro** [comment]
Hi @LindsayMacfarlane - As discussed with @mikevespi, this ticket is ready for you to review and can be closed with updates summarized below:
- CIF perspectives and preferences have be documented above and brought back to the team
- After additional technical discussions (Feb 22nd Backlog Refinements), the design decisions and next steps agreed upon and documented above are **NOT** relevant any more
- Based on the CIF input, @mikevespi, @dleard, and the team agree to follow up on white boarding sessions to come up with technical design solutions, which may have impacts on the workflows
- The implementation of Amendments/Revisions is currently on hold until technical design decisions coming out of the white boarding session.

**Comment by LindsayMacfarlane** [comment]
Thank you @nanyangpro! I appreciate how much work this has been to bring the team and program staff to a common understanding of how this feature can be implemented. I look forward to seeing the next steps.
